localgovdrupal/localgov_forms
# Feature: Long term storage of Webform submission data

## What does this change?

Adds a submodule to store Webform submission data into a separate database for long term storage (LTS).

Personally identifiable information (PII) is optionally redacted before copying into the LTS database.

## How to test
Please see the [Setup](https://github.com/localgovdrupal/localgov_forms/blob/feature/long-term-storage/modules/localgov_forms_lts/README.md#setup-process) and [Inspection](https://github.com/localgovdrupal/localgov_forms/blob/feature/long-term-storage/modules/localgov_forms_lts/README.md#inspection) sections of the README.

## How can we measure success?

Webform submissions should get copied into the LTS database during cron runs.

## Have we considered potential risks?

PII redaction may miss relevant pieces information.  Current redaction rules are as follows:
- All Webform elements of type email, telephone, number, address, localgov_forms_dob.
- Any element with the following happening in its machine id: name, mail, phone, date_of_birth, personal, title, gender, sex, ethnicity, post_code.
- All textareas are cleaned of email, postcode, and any number.  Email detection uses a [simple regex](https://github.com/localgovdrupal/localgov_forms/blob/a431c1c0df294f16b15df1c0c9c888732edbbcbe/modules/localgov_forms_lts/src/PIIRedactorForText.php#L86) which may miss complex email address formats.
